keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -192,10 +178,13 @@ class ApplicationController(
   )
   @ResponseStatus(HttpStatus.ACCEPTED)
   fun deletePin(
+    @RequestHeader("X-SPINNAKER-USER") user: String,
     @PathVariable("application") application: String,
-    @PathVariable("targetEnvironment") targetEnvironment: String
+    @PathVariable("targetEnvironment") targetEnvironment: String,
+    @RequestParam reference: String? = null,
```

Comment:
There's now a single API to delete a pin, with `reference` being optional. If it's missing, all pins for that environment will be deleted, otherwise, only the one corresponding to the artifact reference.

Reply:
Could you remove comment from here? It's not used (I know you'd like to add it back in later).
I generally don't love having comments as URL parameters. It feels weird to me, and they could potentially be long or weird

Reply:
But we do use it... It's the "reason" field in the UI, which I assumed we would add for unpinning too @erikmunson?
![image](https://user-images.githubusercontent.com/1323478/81619072-9a886500-939d-11ea-83b5-3ce2650cf33a.png)

Reply:
P.S. This is where that new event would come in as unpinning currently just deletes a record from the DB and doesn't store any other information for us to query later.

Reply:
To be clear, I'm not suggesting that we don't have a comment. Just that you don't use it in this PR, so it probably doesn't belong here

Reply:
I don't really see any meaningful value in adding a comment to an unpin operation, especially if the only place it will show up is in a history event's details. Probably not worth worrying about supporting it, I doubt we will expose it in the UI.